search-service/src/main/kotlin/com/egm/stellio/search/util/TemporalQueryUtils.kt

```diff
@@ -1,32 +1,33 @@
 package com.egm.stellio.search.util
 
 import arrow.core.*
+import arrow.core.continuations.either
 import com.egm.stellio.search.config.ApplicationProperties
 import com.egm.stellio.search.model.AttributeInstance
 import com.egm.stellio.search.model.TemporalEntitiesQuery
 import com.egm.stellio.search.model.TemporalQuery
+import com.egm.stellio.shared.model.APIException
 import com.egm.stellio.shared.model.BadRequestDataException
 import com.egm.stellio.shared.util.*
-import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_CREATED_AT_PROPERTY
-import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_MODIFIED_AT_PROPERTY
-import com.egm.stellio.shared.util.JsonLdUtils.buildNonReifiedDateTime
 import org.springframework.util.MultiValueMap
 import java.time.ZonedDateTime
 import java.util.Optional
 
-fun parseAndCheckQueryParams(
+suspend fun parseQueryAndTemporalParams(
     pagination: ApplicationProperties.Pagination,
     requestParams: MultiValueMap<String, String>,
     contextLink: String,
     inQueryEntities: Boolean = false
-): TemporalEntitiesQuery {
-    val queryParams = parseAndCheckParams(
+): Either<APIException, TemporalEntitiesQuery> = either {
+    val queryParams = parseQueryParams(
         Pair(pagination.limitDefault, pagination.limitMax),
         requestParams,
         contextLink
-    )
+    ).bind()
+
     if (queryParams.types.isEmpty() && queryParams.attrs.isEmpty() && inQueryEntities)
-        throw BadRequestDataException("Either type or attrs need to be present in request parameters")
+        BadRequestDataException("Either type or attrs need to be present in request parameters")
+            .left().bind<TemporalEntitiesQuery>()
```

Comment:
why add <TemporalEntitiesQuery>

Reply:
I have a warning if not (but don't really know why it can't automatically infer it, we have the same thing in the web handlers)